docs/reference/analysis/tokenfilters/synonym-graph-tokenfilter.asciidoc

```diff
@@ -160,12 +157,50 @@ Text will be processed first through filters preceding the synonym filter before
 {es} will also use the token filters preceding the synonym filter in a tokenizer chain to parse the entries in a synonym file or synonym set.
 In the above example, the synonyms graph token filter is placed after a stemmer. The stemmer will also be applied to the synonym entries.
 
-The synonym rules should not contain words that are removed by a filter that appears later in the chain (like a `stop` filter).
-Removing a term from a synonym rule means there will be no matching for it at query time.
-
 Because entries in the synonym map cannot have stacked positions, some token filters may cause issues here.
 Token filters that produce multiple versions of a token may choose which version of the token to emit when parsing synonyms.
 For example, `asciifolding` will only produce the folded version of the token.
 Others, like `multiplexer`, `word_delimiter_graph` or `ngram` will throw an error.
 
 If you need to build analyzers that include both multi-token filters and synonym filters, consider using the <<analysis-multiplexer-tokenfilter,multiplexer>> filter, with the multi-token filters in one branch and the synonym filter in the other.
+
+[discrete]
+[[synonym-graph-tokenizer-stop-token-filter]]
+===== Synonyms and `stop` token filter
+
+Synonyms and <<analysis-stop-tokenfilter,stop token filter>> interact with each other in the following ways:
+
+[discrete]
+====== Stop token filter *before* synonym token filter
+
+Stop words will be removed from the synonym rule definition.
+This can can cause errors on the synonym rule.
+
+WARNING: An invalid synonym rule will cause an error applying your analyzer changes.
+For reloadable analyzers, the analyzer will not be reloaded and the changes will not be applied.
+You must correct the error in the synonym rules and reload the analyzer, as an index that contains an invalid synonym rule won't be able to be reopened.
+This means that the index will be inoperable when a node that contains the index is started, or the index is opened from a closed state.
```

Comment:
```suggestion
This means that the index will be inoperable when a node that contains the index is (re)started, or the index is opened from a closed state.
```

Reply:
Yes, technically, but the most common way for the index to be opened from a closed state is for the node to be restarted. So maybe good to call that case out more specifically?

Reply:
I'd say it's not needed - lets leave that to the docs team to decide 👍

Reply:
You could add use bullet points to spell out the 3 cases:

```asciidoc
[WARNING]
====
Invalid synonym rules can cause errors when applying analyzer changes.
For reloadable analyzers, this prevents reloading and applying changes.
You must first correct errors in the synonym rules and reload the analyzer.

An index with invalid synonym rules cannot be reopened, making it inoperable when:

* A node containing the index starts
* The index is opened from a closed state
* A node restart occurs (which reopens closed indices)
====
```

But I'll let you folks decide if that's overkill, my only comment would be that if we can meaningfully reduce cognitive load when reading the warning message, then maybe Carlos' original version is OK. I also did a little copyediting to make the original verbiage more concise, feel free to use zero or more of these rephrasings :)